qutebrowser/browser/commands.py

```diff
@@ -1748,3 +1756,19 @@ def fullscreen(self, leave=False, enter=False):
 
         log.misc.debug('state before fullscreen: {}'.format(
             debug.qflags_key(Qt, window.state_before_fullscreen)))
+
+    def possibly_open_in_existing_tab(self, url):
+        if not config.val.tabs.switch_to_open_url:
+            return False
+
+        tab = next((t for t in tabsfinder.all_tabs() if t.url == url), None)
+
+        if tab is not None:
+            index = "{}/{}".format(tab.win_id, tab.index)
+            self.buffer(index)
```

Comment:
Having to construct a string index which is then parsed by the `:buffer` command again seems odd. Why not add a `switch_to_tab(win_id, tab_id)` or something to `tabsfinder.py` (or `tabutils.py`), then use that from `:buffer` and also use it from here?

Reply:
Done.